lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -21,6 +22,17 @@ class ListTools(ToolCommand):
     def __init__(self, context):
         super(ListTools, self).__init__(context)
 
+    @staticmethod
+    def print_results(toolinfo: dict, with_options: bool):
+        for group in sorted(toolinfo.keys()):
+            for host in sorted(toolinfo[group].keys()):
+                tools = toolinfo[group][host]
+                if not with_options:
+                    s = ", ".join(tools)
+                else:
+                    s = ", ".join(map(lambda x: " ".join(x), tools))
```

Comment:
I noticed that below you sort the list of groups you're printing; makes me wonder if sorting here would also help to make the output more predictable and perhaps a bit easier to read. (Sorting the options as well as the tools themselves, perhaps?)

Reply:
The iterators at lines 27 & 28 are sorted...does that not do what you're hoping?  The list itself is sorted at line 100 because it's not using a sorted iterator like it does at line 96.

Reply:
The group and host iterations are sorted, but not the tools for each host, nor the options for each tool. Sorting the tools would make the captured output in the test a bit easier to sort out; sorting the options may be less interesting, but might still help when comparing rows.

In any case I approved, so this is just extra feedback to think about. To emphasize, I'm going to make this harder to follow by marking the conversation resolved. 😆

Reply:
OK, I see what you mean.  (And, yes, based on recent experiences with unit tests, sorting them would remove a source of future surprise....)